libs/@blockprotocol/graph/src/shared/stdlib/subgraph/element/entity.ts

```diff
@@ -36,6 +36,37 @@ export const getEntities = <Temporal extends boolean>(
   });
 };
 
+const getRevisionsForEntity = <Temporal extends boolean>(
+  subgraph: Subgraph<Temporal>,
+  entityId: EntityId,
+): Vertices<Temporal>[string] | undefined => {
+  const entityRevisions = subgraph.vertices[entityId];
+
+  if (entityRevisions) {
+    return entityRevisions;
+  }
+
+  // check for the presence of draft versions of the entity in the subgraph
+  const draftEntityIds = Object.keys(subgraph.vertices).filter((id) =>
+    id.startsWith(`${entityId}~`),
+  );
```

Comment:
If `entityId` already contains two `~` we can return `undefined` here as an early return. Small optimization which I don't know worth implementing it.

Reply:
Might as well! Done in https://github.com/blockprotocol/blockprotocol/pull/1374/commits/7c12c757a1967935a7b5c5244fc7053a8e0e758f